### ISSUE TITLE
MacOS packaged app crash after signing “$APP Helper (GPU).app” with security.app-sandbox

### ISSUE THREAD

**Issue by le1771** [electron-userland/electron-builder#4899]
* **Version**:  22.5.1

* **Electron Version**: 8.2.3
* **Electron Type (current, beta, nightly)**: current

* **Target**: MacOS

I've built an app with ElectronJS and electron-builder.
The app has been working well on Win, Linux, so I'm expanding to release for MacOS.

I'm running in issues in signing the app. After signing, the app crashed immediately when launched. (attached crashlog).
[Electron Quick Start Helper (GPU)_2020-04-25-154511.crash.zip](https://github.com/electron-userland/electron-builder/files/4534029/Electron.Quick.Start.Helper.GPU._2020-04-25-154511.crash.zip)
I'm using MacOS 10.15.4

So I've clone **electron-quick-start** and apply the same signing steps, and get the same issue.
After many trial and errors.
I've narrowed down the issue to that if we sign:
`codesign -s "$APP_KEY" -f --entitlements "$CHILD_PLIST" "$FRAMEWORKS_PATH/$APP Helper (GPU).app/"`
where the PLIST has 
`<key>com.apple.security.app-sandbox</key><true/>`
The app will crash.

**Steps to replicate:**
You can try to clone my fork of electron-quick-start here:
https://github.com/le1771/electron-quick-start

First run:
```
export CSC_IDENTITY_AUTO_DISCOVERY=false. #to disable auto sign
yarn
yarn package-mac
```

Run the packaged "Electron Quick Start.app" in /release/mas
=> it launched fine

Next run the signing script, you will need to set value for $APP_KEY, $INSTALLER_KEY
`./build/resignAndPackage.sh`

Now run the "Electron Quick Start.app" in /release/mas again, it crashes.

If I remove the 
`<key>com.apple.security.app-sandbox</key><true/>`
in entitlements.mas.inherit.plist then the app will not crash.
But of course, we can't notarize the app or submit to Apple Store because it lack of sandbox.

If I add app-sandbox back, the signing works well, and pass Apple's Notary check, and I was able to upload to Apple App Store, but of course, it failed review because the app crashed on start up.

I've tried with various combination of nodejs, electron and electron builder, all have the same issue.
What I tried:
- node: 12.16.2 - electron 8.2.3 - electron-builder 22.5.1
- node: 10.20.1 - electron 8.2.3 - electron-builder 22.5.1
- node: 10.20.1 - electron 8.2.3 - electron-builder 21.2.0
- node 10.20.1 - electron 6.1.10 - electron-builder 22.5.1
- node 10.20.1 - electron 6.1.7 - electron-builder 22.5.1
- node 8.17.0 - electron 4.2.12 - electron-builder 22.5.1
- node 8.17.0 - electron 4.2.12 - electron-builder 21.2.0
- node 8.17.0 - electron 4.2.12 - electron-builder 20.44.4
The older electron 4.2.12 did not come with $APP Helper (GPU).app, it only come with $APP Helper.app, but after signing the $APP Helper.app, the app crashed the same way.

I've referred to:
https://www.electronjs.org/docs/tutorial/mac-app-store-submission-guide
https://medium.com/@jondot/shipping-electron-apps-to-mac-app-store-with-electron-builder-e960d46148ec
https://kilianvalkhof.com/2019/electron/notarizing-your-electron-application/

Has anyone run into this issue before? Any help would be much appreciated!

**Comment by dmtask** [comment]
I have the same problem, but apparently it's not so easy to solve, because the electron-builder developers haven't released an update yet. But the problem is currently covered in a lot of issues here, but without success so far.

**Comment by le1771** [comment]
I also tried to package the electron-quick-start app with electron-packager.
Then sign with the same process, and get the same issue.
So could this be issue with ElectronJS itself?

But I've also saw in other issues some have successfully published ElectronJS app to MacOS app store, could this be changes recently?

**Comment by dmtask** [comment]
> So could this be issue with ElectronJS itself?
Good question, I don't think because electron itself has nothing to do with the build process, right?

> ElectronJS app to MacOS app store, could this be changes recently?
I have asked myself the same question. But I think that there are only problems with developer certificates. As soon as a correct/real certificate is taken, the problems are no longer there.

**Comment by le1771** [comment]
Thanks Daniel, did you able to sign successfully with different certificate?

**Comment by dmtask** [comment]
No. I try it with the Developer ID Application Certificate.

**Comment by wgcv** [comment]
I think that the problem is plist are in binary and not XML check https://github.com/electron/electron-osx-sign/issues/223#issuecomment-611070794

**Comment by le1771** [comment]
Thanks @wgcv, the .plist filerelease/mas/Electron Quick Start.app/Contents/Frameworks/Electron Quick Start Helper (GPU).app/Contents/Info.plist
and other Framework .plist files are indeed were packaged in binary format as the post mentioned.
I've used the command

plutil -convert xml1 "release/mas/Electron Quick Start.app/Contents/Frameworks/Electron Quick Start Helper (GPU).app/Contents/Info.plist"

But the app still crashed immediately like before.
I've tried to covert XML for all the plist, still same issue.
